marquez/client.py

```diff
@@ -0,0 +1,238 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import collections
+import os
+import urllib
+
+import requests
+from marquez.constants import JobRunState
+from marquez.utils import (InvalidClientConfigurationError,
+                           InvalidRequestError, _compose_path,
+                           handle_response)
+
+
+class Client(object):
+    API_PATH = "api/v1"
+    DEFAULT_TIMEOUT_MS = 5000
+    api_base = None
+    namespace = None
+
+    def __init__(self, host=None, port=None, timeout_ms=None):
+        host = os.environ.get('MARQUEZ_HOST') or host
+        port = os.environ.get('MARQUEZ_PORT') or (
+            str(port) if port else port)
+        t = os.environ.get('MARQUEZ_TIMEOUT_MS') or str(timeout_ms)
+        timeout_milliseconds = (int(t) if t.isnumeric() else
+                                self.DEFAULT_TIMEOUT_MS)
+        self.timeout = float(timeout_milliseconds / 1000)
+
+        if not host or not port:
+            msg = ("Please provide host & port or set the proper env "
+                   "vars: MARQUEZ_HOST, MARQUEZ_PORT")
+            raise InvalidClientConfigurationError(msg)
+
+        self.api_base = "http://{0}:{1}/{2}".format(
```

Comment:
We'll want to make `api_base` private: `self.api_base` **->** `self._api_base`

Reply:
Done.